Modules/syslogmodule.c

```diff
@@ -162,8 +170,9 @@ syslog_openlog_impl(PyObject *module, PyObject *ident, long logopt,
     }
 
     openlog(ident_str, logopt, facility);
```

Comment:
This API is stateless. What happens if interpreter A calls openlog("idA", options_a, LONG_USER) and interpreter B calls ``openlog("idB", options_b, LOG_INFO)``?

I'm not sure if it's a good idea to make the global ``S_log_open`` flag per-interpreter. See for example the recent crash on macOS: #98178. Same question for S_ident_o.

If you're confident that safe, please add a comment explaining the behavior, who owns the memory, etc.

Reply:
It is safe, but the behavior might be unexpected.  The only issue of safety would be how POSIX syslog keeps a pointer to "ident" that must not be deallocated before `closelog()` is called.  I don't see where that would be a problem.

That said, moving the data to per-interpreter module state may lead to a slight change in behavior.  Again, though, using syslog with multiple interpreters is already wierd.